### DISCUSSION TITLE
Maven POM Dependency issues

### DISCUSSION THREAD

**Discussion by sprince1504** [graphql-java/graphql-java#3210] [General]
Hi Team,

I am working on an AEM-Java project and facing an issue with graphql-java dependency where I am trying to add it on a maven project having a BND plugin. but once I add the dependency and try to build the code so the OSGI bundle is always in the installed state. I am not sure what I missing but looking like there is an other jar dependency on graphql-java.

I also raised this request on Adobe AEM Community forum but no one was able to help. can anyone please help me here?

https://experienceleaguecommunities.adobe.com/t5/adobe-experience-manager/aem-graphql-java/m-p/590600#M147441

Thank you,
Prince

**Comment by bbakerman** [comment]
I am going to move this to a discussion - its not an issue with graphql-java itself

**Reply by sprince1504** [reply]
@bbakerman  - Thanks for moving it to the discussion. Could you please help me here?

**Comment by sprince1504** [comment]
Update: I am getting this message while trying to add the dependency:

[ERROR] [api-regions-exportsimports] com.adobe.aem.guides.aemspa.react:aem-guides-aem-spa.react.core:1.0.0-SNAPSHOT: Bundle aem-guides-aem-spa.react.core:1.0.0-SNAPSHOT is importing package(s) [com.google.errorprone.annotations, org.reactivestreams, org.dataloader.stats, org.checkerframework.checker.nullness.qual, org.dataloader, com.google.errorprone.annotations.concurrent, javax.annotation.meta] in start level 20 but no bundle is exporting these for that start level. (com.adobe.aem.guides.aemspa.react:aem-guides-aem-spa.react.all:1.0.0-SNAPSHOT)

Added dependency in core pom.xml and parent pom.xml with version 20.2 and tried 19 as well.

Added this in core pom.xml - 
Import-Package: javax.annotation;version=0.0.0,!com.google.,!org.checkerframework.,!javax.annotation.,!graphql.com.google.,!sun.misc,*
-includeresource: graphql-java-[0-9.]*.jar;lib:=true

**Reply by bbakerman** [reply]
Sorry this is a OSGI problem - I am not someone who can help with it.

We have people using graphql-java in OSGI land - so maybe some one out there might help

**Comment by renatoathaydes** [comment]
It seems to me this is due to the graphql-java embedding Guava (and its dependencies). That's where package `com.google.errorprone.annotations` comes from (this is a compile-time only dependency of Guava which should probably be excluded when graphql-java shades Guava).

If I understand correctly, `reactivestreams` is an optional dependency of graphql-java. @sprince1504 if you explicitly tell the runtime to not import those packages, everything should work fine (as long as you don't hit the code using reactor, which I think you only hit if you explicitly use that in your code).

**Comment by sprince1504** [comment]
Hi All,

The issue is resolved. I posted this on StackOverflow and got the solution.
https://stackoverflow.com/questions/76233122/maven-pom-dependency-issues-3210?noredirect=1#comment134487966_76233122
Thank you for your support.
